playbooks/roles/foreman_libvirt/tasks/foreman.yml

```diff
@@ -0,0 +1,186 @@
+  register: foreman_libvirt_compute_resource
+  ignore_errors: True
+
+- name: 'create compute resource'
+  shell: >
+    {{ foreman_libvirt_hammer }} compute-resource create
+    --name "libvirt"
+    --url "qemu:///system"
+    --provider libvirt
+    {% if foreman_libvirt_organizations %}
+    --organizations "{{ foreman_libvirt_organization }}"
+    {% endif %}
+    {% if foreman_libvirt_locations %}
+    --locations "{{ foreman_libvirt_location }}"
+    {% endif %}
+  when: foreman_libvirt_compute_resource.stderr.find('not found') != -1
+
+# Domain
+- name: 'find domain'
+  shell: >
+    {{ foreman_libvirt_hammer }} domain info --name "example.com"
+  register: foreman_libvirt_domain
+  ignore_errors: True
+
+- name: 'create domain'
+  shell: >
+    {{ foreman_libvirt_hammer }} domain create
+    --name example.com
+    --dns-id {{ foreman_libvirt_smart_proxy.Id }}
+    {% if foreman_libvirt_organizations %}
+    --organizations "{{ foreman_libvirt_organization }}"
+    {% endif %}
+    {% if foreman_libvirt_locations %}
+    --locations "{{ foreman_libvirt_location }}"
+    {% endif %}
+  when: foreman_libvirt_domain.stderr.find('not found') != -1
+
+- name: 'update domain'  # it may have been automatically created by puppet if katello reports first
+  shell: >
+    {{ foreman_libvirt_hammer }} domain update
+    --name example.com
+    --dns-id {{ foreman_libvirt_smart_proxy.Id }}
+    {% if foreman_libvirt_organizations %}
+    --organizations "{{ foreman_libvirt_organization }}"
+    {% endif %}
+    {% if foreman_libvirt_locations %}
+    --locations "{{ foreman_libvirt_location }}"
+    {% endif %}
+
+# Subnet
+- name: 'find subnet'
+  shell: >
+    {{ foreman_libvirt_hammer }} subnet info --name "192.168.73.0/24"
+  register: foreman_libvirt_subnet
+  ignore_errors: True
+
+- name: 'create subnet'
+  shell: >
+    {{ foreman_libvirt_hammer }} subnet create
+    --name "192.168.73.0/24"
+    --dhcp-id {{ foreman_libvirt_smart_proxy.Id }}
+    --dns-id {{ foreman_libvirt_smart_proxy.Id }}
+    --tftp-id {{ foreman_libvirt_smart_proxy.Id }}
+    --domains example.com
+    --from 192.168.73.2
+    --to 192.168.73.100
+    --network 192.168.73.0
+    --mask 255.255.255.0
+    --ipam DHCP
+    --gateway 192.168.73.1
+    --dns-primary 192.168.73.1
+    {% if foreman_libvirt_organizations %}
+    --organizations "{{ foreman_libvirt_organization }}"
+    {% endif %}
+    {% if foreman_libvirt_locations %}
+    --locations "{{ foreman_libvirt_location }}"
+    {% endif %}
+  when: foreman_libvirt_subnet.stderr.find('not found') != -1
+
+# Puppet
+- name: 'find environment'
+  shell: >
+    {{ foreman_libvirt_hammer }} environment info --name "production"
+  register: foreman_libvirt_environment
+  ignore_errors: True
+
+- name: 'create environment'
+  shell: >
+    {{ foreman_libvirt_hammer }} environment create
+    --name production
+    {% if foreman_libvirt_organizations %}
```

Comment:
I guess thats nice for some classes of users, sure would be nice to just have a single behavior to reduce branching.